### PR TITLE
NP-2492: Add support for telegraf-0.10

### LINKED ISSUE #7: NP-2492 Support telegraf 0.10.x
telegraf 0.10.x has brought some breaking changes.  We need to update the cookbook to support those changes.  _Note:_ the database format also changed, but migrating databases should be out of scope.

### PR DESCRIPTION
**Pull request by tenyo**
This PR adds support for the new version of telegraf (0.10) and will no longer work with 0.2.x. Resolves #7 
- add new resource telegraf_inputs; remove telegraf_plugins
- change default installed version to 0.10.0-1
- add unstable apt repository source for ubuntu

Note that Telegraf 0.10 also changes the format of the output stats, so if used with Grafana or similar tool you'll need to re-map the new fields in all of your graphs.

For more details about the changes in telegraf 0.10:
https://influxdata.com/blog/announcing-telegraf-0-10-0/

### CODE REVIEW

**Comment by fishnix** [comment]
Although I appreciate the idea, I don't think there's any reason to continue supporting the 0.2.x in this cookbook.  We cut a release so anyone wanting to run the old version can run the old release.  Also, can you update the serverspec tests?  They still point to the old paths, etc.

File: attributes/default.rb [hunk 1]
```diff
@@ -17,7 +17,14 @@
...
-default['telegraf']['config_file_path'] = '/etc/opt/telegraf/telegraf.conf'
+# version of telegraf to install, e.g. '0.10.0-1', '0.2.4-1' or nil for the latest
+default['telegraf']['version'] = '0.10.0-1'
+
+if node['telegraf']['version'].start_with? '0.2.'
+  default['telegraf']['config_file_path'] = '/etc/opt/telegraf/telegraf.conf'
+else
+  default['telegraf']['config_file_path'] = '/etc/telegraf/telegraf.conf'
+end
 default['telegraf']['config'] = {
```
Comment (fishnix) [inline attributes/default.rb:23]: The whitespace/newlines is out of whack in this file (not all from this PR), can you clean that up?  I prefer no newline between default attributes but you can add newlines around the conditionals.  Thanks!

**Comment by tenyo** [comment]
Yeah, I had serious doubts about that approach - it was getting kinda messy and could've caused some headaches down the road. I'll update it to only work with the 0.10 telegraf. If it's a major release that should be fine.

**Comment by fishnix** [comment]
I was surprised to see the apt package was suffixed with `-1` but it seems to work properly.  looks good! :+1:

**Comment by tenyo** [comment]
Looks like they made the versions consistent across apt/yum. Hopefully they won't change that in the next release.

**Pull request merged**